### DISCUSSION TITLE
Dil destegi (Q&A translation)

### DISCUSSION THREAD

**Discussion by bersa2009** [cedricp/ddt4all#979] [Q&A]
Program kurukurken Türkçe dil desteği yoktur 😔 lütfen türkçe dil destegi bırak

**Comment by bersa2009** [comment]
[Bersa2009](https://github.com/bersa2009) yorum yaptı [şimdi](https://github.com/cedricp/ddt4all/issues/977#issue-2058718872)
Program kurukurken Türkçe dil desteği yoktur 😔 lütfen türkçe dil destegi bırakın

**Reply by TA1GI** [reply]
Türkçe dil desteği var. Çeviri bana ait. Normalde windows'un varsayılan dili ile açılması gerekiyor. Kurulum aşamasında yaptığın dil seçimi sadece kurulum aşamalarına ait. Daha önce aşağıdaki konuda sorunu bildirdim. orada çözümden de bahsettim. 
Çözüm burada: https://github.com/cedricp/ddt4all/issues/920

**Comment by Furtif** [comment]
Support is in https://github.com/cedricp/ddt4all/tree/master/dtt4all_data/locale/tr/LC_MESSAGES

**Comment by bersa2009** [comment]
While installing the program, there is no Turkish language in the exe file. When you leave it later, Turkish language cannot be translated... When you download the Turkish package from githup.com, you leave it in the c// folder. When you open the program, there is no Turkish language... When you download the source files from githup.com, there is a Turkish file. These source files are exe It would be easier if there was a translated file, if there was a Turkish language test when installing… I cannot change the language because there is no other explanation.

**Comment by bersa2009** [comment]
Could you please send me the exe file via e-mail? 🙏 Turkish language support..

**Comment by bersa2009** [comment]
When loading the exe file, it shows 6-7 or maybe more languages ​​and there is no Turkish in it. Can you send me a ready exe file in Turkish for Windows 10 64 bit computer?

**Comment by Furtif** [comment]
uses https://poedit.net/

**Reply by bersa2009** [reply]
I installed it, what do I do now

**Reply by Furtif** [reply]
open https://github.com/cedricp/ddt4all/blob/master/dtt4all_data/locale/tr/LC_MESSAGES/ddt4all.po and update base from 
https://github.com/cedricp/ddt4all/blob/master/dtt4all_data/locale/ddt4all.pot

![image](https://github.com/cedricp/ddt4all/assets/11375035/34d12356-19bb-4984-8f45-8bc1e9447919)

after click in lines 
![image](https://github.com/cedricp/ddt4all/assets/11375035/bffa3332-8618-4041-985f-6d4b50935c61)

translate it and after end compile 

![image](https://github.com/cedricp/ddt4all/assets/11375035/897b238b-3717-42c9-ac27-bd35b233a4ae)

save as MO replace this https://github.com/cedricp/ddt4all/blob/master/dtt4all_data/locale/tr/LC_MESSAGES/ddt4all.mo

**Reply by Furtif** [reply]
And save new translated also as po https://github.com/cedricp/ddt4all/blob/master/dtt4all_data/locale/tr/LC_MESSAGES/ddt4all.po

**Reply by Furtif** [reply]
When you do the translation, make a Pull Request (PR) with the proposal of the two files, if you don't know how to make a PR you can use Discord and send the files there to replace them in the repo, the link to discord you have in the badge in the readme file https://github.com/cedricp/ddt4all?tab=readme-ov-file#ddt4all--- or https://discord.gg/cBqDh9bTHP

**Reply by Furtif** [reply]
i have updated base now you need only edit and fine.

**Reply by TA1GI** [reply]
There is no problem with the translation file. I explained what the real problem is here: https://github.com/cedricp/ddt4all/issues/920

**Reply by Furtif** [reply]
@TA1AUB Changed mode https://github.com/cedricp/ddt4all/blob/08eac1a059b8248901d745aeca3f353fdb3ac530/options.py#L42 maybe ok now

**Reply by TA1GI** [reply]
@Furtif  Yes, I tested that too, but the same problem still persists.

**Reply by Furtif** [reply]
@TA1AUB I have to add a Language->List menu for manual choice in case of problem

**Reply by TA1GI** [reply]
@Furtif If line 45 is changed as follows, the problem is solved:
lang, enc = locale.getdefaultlocale()

**Reply by Furtif** [reply]
is odd maybe tr as more filials more or les style US_UK US_US FR_FR_ FR_CH FR_BE etc... now
need look this
```
Possible translations:
English code: en_US
German code: de
Spanish code: es
French code: fr
Hungarian code: hu
Italian code: it
Dutch code: nl
Polish code: pl
Portuguese code: pt
Romanian code: ro
Russian code: ru
Serbian code: sr
Turkish code: tr
Ukrainian code: uk_UA
```

Edit configuration `dtt4all_data/config.json` as you code. Ex: "lang": "en_US"
Edit it only if it not ok for you country language.```

Look this:
https://github.com/cedricp/ddt4all/pull/982/commits/018c3fd0593a9754861dec73e2c30e6d4a61f38c

**Reply by Furtif** [reply]
ex 
![image](https://github.com/cedricp/ddt4all/assets/11375035/06569a99-f023-46a3-8b32-aa3826923694)

## But for now not ok this menu 
- Edit you self config.json at:
```json
{
 "lang": "tr"
}

**Reply by Furtif** [reply]
test edit json, if ok go make better

**Reply by Furtif** [reply]
@TA1AUB ... i need you test

**Reply by Furtif** [reply]
use branch https://github.com/cedricp/ddt4all/tree/py_3.12.x

**Reply by TA1GI** [reply]
@Furtif  I can't find this file.dtt4all_data/config.json

**Reply by TA1GI** [reply]
@Furtif    I tested this branch without any changes. It did not open with the default system language. It opened in English. 
I tested it by selecting Turkish language from the interface language section. It opened in English.
https://github.com/cedricp/ddt4all/tree/py_3.12.x

**Reply by TA1GI** [reply]
@Furtif  After running the program, I noticed that the dtt4all_data/config.json file was created. Now, when I edited the file as follows, the program opened in Turkish.
{
  "lang": "tr_TR"
}

But when I selected the interface language as English, it opened in Turkish.

**Reply by Furtif** [reply]
@TA1AUB edit config
set tr 
``` 
{
 "lang": "tr"
}
```
![image](https://github.com/cedricp/ddt4all/assets/11375035/0ee115b9-a124-4f74-a699-bd236d3eb44e)

**Reply by TA1GI** [reply]
@Furtif I think we posted comments at the same time.

**Reply by Furtif** [reply]
now for auto you says "tr_TR" but tr is tr only you need douplicate or rename te to tr_TR folder

**Reply by TA1GI** [reply]
Well then let's just do it like you said.
{
  "lang": "tr"
}

**Reply by Furtif** [reply]
reload all new commits delete config.json reload ... test again

**Reply by TA1GI** [reply]
@Furtif  I tested with clean install and py_3.12.x branch. The content of the config.json file is as follows:
{
  "lang": "Turkish_Turkey"
}
And it opened in English.

When I edited the file as follows, it opened in Turkish.
{
  "lang": "tr"
}

**Reply by Furtif** [reply]
"lang": "Turkish_Turkey" is bah need be tr only

**Reply by Furtif** [reply]
{
 "lang": "tr"
}

**Reply by Furtif** [reply]
or
{
 "lang": "tr_TR"
}

**Reply by Furtif** [reply]
config as tr only is this 

![image](https://github.com/cedricp/ddt4all/assets/11375035/9d58bd75-444b-47a1-9ec5-2e7814bcc5d2)

**Reply by Furtif** [reply]
the only thing that you have extremely bad in your translation is the title of the application
should have preserved it

**Reply by TA1GI** [reply]
@Furtif You are right. I am sorry. I will fix it. But in the first translation added, there was a website in the application title.
https://github.com/TA1AUB/ddt4all/commit/7546dc56752c7acefded6fa001d79303fb541572

**Reply by Furtif** [reply]
@TA1AUB you have try last commits?

**Reply by TA1GI** [reply]
@Furtif  I tested the final version of the master branch. It works fine. I'm adding a screenshot. Can you add the texts I marked to the translation file?
![Adsız](https://github.com/cedricp/ddt4all/assets/11720438/c4f7924f-44a0-41fb-9b94-0f0de4d67bd4)

**Reply by Furtif** [reply]
I put these logs in reference I thought about removing them but I think I'll leave them, I have to add the translations

**Reply by Furtif** [reply]
I don't know if it will be good, because these reference logs are in options.py, I don't know if it will work, we'll see
https://github.com/cedricp/ddt4all/blob/7d106179aa108c6e53b2e1840b3b67e5da90c42f/options.py#L63

**Reply by Furtif** [reply]
This informative reference only appears if the config.json file does not exist or is poorly structured, after which it no longer appears. If you want to test this, delete the file to get this reference back, I'll try to add it.
